### PR TITLE
Allowing Access-Control-Allow-Headers to always exist in response.

### PR DESCRIPTION
**Pull request by danbev**
Motivation:
A preflight request can benefit from returning all headers that it
allows a client to send, since these headers will be stored in the
preflight cache (similar to how the request methods are stored)
which can potentially save a round trip for subsequent request
that send other/more headers.

This check will allow both empty 'Access-Control-Allow-Headers' or
contain a list of headers that the server allows.

### CODE REVIEW

**Comment by danbev** [comment]
@brycekahle I've rebased and fixed the conflict for this PR. Would you be able review?

File: sockjs-protocol-0.3.3.py [hunk 1]
```diff
@@ -819,21 +819,30 @@ def test_request_headers_cors(self):
                 headers={'Origin': 'test', 'Access-Control-Request-Method': 'POST', 'Access-Control-Request-Headers': 'a, b, c'})
         self.assertTrue(r.status == 204 or r.status == 200)
         self.verify_cors(r, 'test')
-        self.assertEqual(r['Access-Control-Allow-Headers'], 'a, b, c')
+        # A preflight request can benefit from returning all headers that it allows a client to send, since these
+        # headers will be stored in the preflight cache (similar to how the request methods are stored) which can 
+        # potentially save a round trip for subsequent request that send other/more headers.
+        self.assertEqual(r['Access-Control-Allow-Headers'].strip(), 'a,b,c')
```
Comment (brycekahle) [inline sockjs-protocol-0.3.3.py]: Why the change in spacing? I'm not sure this will pass because `strip` removes leading and trailing whitespace, not within the string.

**Comment by danbev** [comment]
@brycekahle Sorry for my late response. You are right, this was wrong and I've updated it now with what I hope will deal with both cases (space after comma and no space after comma)

File: sockjs-protocol-0.3.3.py [hunk 2]
```diff
@@ -819,21 +819,36 @@ def test_request_headers_cors(self):
                 headers={'Origin': 'test', 'Access-Control-Request-Method': 'POST', 'Access-Control-Request-Headers': 'a, b, c'})
         self.assertTrue(r.status == 204 or r.status == 200)
         self.verify_cors(r, 'test')
-        self.assertEqual(r['Access-Control-Allow-Headers'], 'a, b, c')
+        # A preflight request can benefit from returning all headers that it allows a client to send, since these
+        # headers will be stored in the preflight cache (similar to how the request methods are stored) which can 
+        # potentially save a round trip for subsequent request that send other/more headers.
```
Comment (3rd-Eden) [inline sockjs-protocol-0.3.3.py]: I thought that all requests that the SockJS client did had cache busting query strings. Making it impossible to actually cache these responses.

**Comment by danbev** [comment]
@brycekahle Would you mind taking another look at this one? It is the last issue for getting the full test suite to pass against our Netty server (except for a know issue with haproxy test).

I've run these changes against sockjs-node and they pass, but I'm seeing a few websocket test errors, for example:
```shell
$ ./venv/bin/python sockjs-protocol.py WebsocketHybi10.test_transport
E
======================================================================
ERROR: test_transport (__main__.WebsocketHybi10)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "sockjs-protocol.py", line 652, in test_transport
    ws = WebSocket8Client(trans_url)
  File "/Users/danielbevenius/work/sockjs/sockjs-protocol/utils.py", line 99, in __init__
    self.client.connect()
  File "/Users/danielbevenius/work/sockjs/sockjs-protocol/venv/lib/python2.7/site-packages/ws4py/client/__init__.py", line 209, in connect
    self.sock.connect(self.bind_addr)
  File "/usr/local/Cellar/python/2.7.6/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 61] Connection refused

----------------------------------------------------------------------
Ran 1 test in 0.006s

FAILED (errors=1)
```
I started the nodejs server like this:
```shell
$ make test_server
./node_modules/.bin/coffee -o lib/ -c src/*.coffee
node tests/test_server/server.js
SockJS v0.3.17 bound to "/echo"
SockJS v0.3.17 bound to "/disabled_websocket_echo"
SockJS v0.3.17 bound to "/cookie_needed_echo"
SockJS v0.3.17 bound to "/close"
SockJS v0.3.17 bound to "/ticker"
SockJS v0.3.17 bound to "/amplify"
SockJS v0.3.17 bound to "/broadcast"
 [*] Listening on 0.0.0.0:8081
```
Any idea if this is a configuration issue on my part of something?
